---
title: "Taxonomy assignment with pimba_run"
layout: archive
permalink: /handsOn_run/
---  

The output generated by PIMBA_prepare is a fasta file that PIMBA_run will use.

You must configure the inputs for the "run" mode according to the parameters below.

| Parameter | Description |
| ----------- | ----------- |
| outputrun            | Name of the output folder to be generated.                                                        |
| strategy             | Analysis strategy to be used. Can be "otu" or "asv". If "otu", PIMBA uses vsearch. If "asv", swarm is used. |
| otu_similarity       | Percentage of similarity used in OTU clustering. The default is 0.97.                            |
| assign_similarity    | Percentage of similarity used in taxonomy assignment. The default is 0.9.                       |
| mincoverage          | Minimum alignment coverage. The default is 0.9.                                              |
| otu_length           | Minimum length for trimming reads. If the value is 0, no reads will be trimmed.                 |
| hits_per_subject     | If 1, choose the best hit. If > 1, choose by majority. The default is 1.                        |
| marker_gene          | Marker gene and database for the analysis. Can be: 16S-SILVA, 16S-GREENGENES, 16S-RDP, 16S-NCBI, ITS-FUNGI-NCBI, ITS-FUNGI-UNITE, ITS-PLANTS-NCBI, COI-BOLD, or COI-NCBI. |
| e_value              | Expected value (e-value) used by BLAST. The default is 0.001.                                  |
| lulu                 | If set to 'yes', PIMBA will discard erroneous OTUs or ASVs using LULU. The default is 'no' (not using LULU). |
| ITS                  | Set to 'yes' if the reads are ITS.                                                               |
| remote | Define whether BLAST will be done in remote mode (without downloading the database) or local mode. |


#### Database paths
Depending on the database used, provide the full path to the files related to that database in the `pimba_smk/config/config.yaml`. Snakemake will only use the path specified in the "marker_gene" option from the previous item, so only the specified marker gene needs to be configured.

If you have chosen the COI dataset, just set the `marker_gene` variable in the config file to `COI-BOLD` and set the `COI-BOLD-DB` variable to `/home/program2/db/PIMBA_smk/3/COI_BOLD/`.
If you have chosen the ITS dataset, just set the `marker_gene` variable in the config file to `/home/program2/db/PIMBA_smk/3/ITS_plants/`.

After everything is correctly set in the `config.yaml` file, you will use the command below to run PIMBA_run mode, assuming you are analyzing the COI dataset:

```console
./pimba_smk_main.sh -p no -r COI-BOLD -g no -t $SLURM_CPUS_PER_TASK -c config/config.yaml
```

If you are analyzing the ITS dataset, the command will be:
```console
./pimba_smk_main.sh -p no -r /home/program2/db/PIMBA_smk/3/ITS_plants/ -g no -t $SLURM_CPUS_PER_TASK -c config/config.yaml
```

But remember! You cannot run the commands above directly in the terminal. Copy the file `/home/program2/src/PIMBA_smk/scripts/pimba_run.slurm` to your folder. Open the file after you have pasted it to your work directory to see if everything is correctly set. Then, submit the job to the processing queue:

```console
sbatch pimba_run.slurm
```

The taxonomy assignment will be generated at the `results/01_run/output/AllSamples_otus_tax_assignments.txt`
 
  
